include/scea/algo/pr.hpp

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: BSD-2-Clause
+// Copyright (c) 2023. University of Texas at Austin. All rights reserved.
+
+#pragma once
+
+#include <vector>
+#include <cmath>
+#include <atomic>
+#include <algorithm>
+
+#include "algo_interface.hpp"
+#include "galois/LargeArray.h"
+#include "galois/AtomicHelpers.h"
+#include "galois/Reduction.h"
+
+namespace scea::algo {
+
+class PageRank : public Algo {
+  static constexpr double DAMPING_FACTOR = 0.85;
+  static constexpr int MAX_ITERATIONS    = 100;
+  static constexpr double TOLERANCE      = 1.0e-4;
+
+public:
+  PageRank() = default;
+
+  static void compute(scea::graph::MutableGraph& g) {
+    const uint64_t numNodes = g.size();
+    std::vector<std::atomic<double>> newRank(numNodes);
+    std::vector<double> rank(numNodes, 1.0 / numNodes);
+    std::vector<unsigned> out_degrees(numNodes, 0);
+
+    galois::do_all(
+        galois::iterate(0ul, numNodes),
+        [&](uint64_t i) {
+          g.for_each_edge(i, [&](uint64_t const&) { out_degrees[i]++; });
+        },
+        galois::no_stats(), galois::loopname("ComputeOutDegrees"),
+        galois::steal());
+
+    for (int iter = 0; iter < MAX_ITERATIONS; ++iter) {
+      std::for_each(newRank.begin(), newRank.end(),
+                    [](std::atomic<double>& n) { n.store(0.0); });
```

Comment:
Why `std::for_each` and not `galois::do_all`?

Reply:
Changed.